library/pkparse.c

```diff
@@ -958,7 +1128,11 @@ static int pk_parse_key_sec1_der(mbedtls_ecp_keypair *eck,
                                          MBEDTLS_ERR_ASN1_LENGTH_MISMATCH);
             }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+            if ((ret = pk_get_ecpubkey(&p, end2, pk)) == 0) {
+#else
             if ((ret = pk_get_ecpubkey(&p, end2, eck)) == 0) {
+#endif
```

Comment:
As in other places comments are added I suggest to add also here.
```suggestion
#endif /* MBEDTLS_USE_PSA_CRYPTO */
```